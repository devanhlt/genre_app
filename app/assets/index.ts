const AppImgs = {
  splashBg: require("./images/splashscreen/splashscreen.png"),
  appLogo: require('./images/logo/logo.png'),
}

export { AppImgs }
