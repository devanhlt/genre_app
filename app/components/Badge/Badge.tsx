import * as React from "react"
import { Animated, StyleProp, StyleSheet, TextStyle, useWindowDimensions } from "react-native"

const defaultSize = 20

export type Props = React.ComponentProps<typeof Animated.Text> & {
  /**
   * Whether the badge is visible
   */
  visible?: boolean
  /**
   * Content of the `Badge`.
   */
  children?: string | number
  /**
   * Size of the `Badge`.
   */
  size?: number
  style?: StyleProp<TextStyle>
  textColor?: TextStyle["color"]
  backgroundColor?: TextStyle["backgroundColor"]
  borderRadius?: TextStyle["borderRadius"]
  ref?: React.RefObject<typeof Animated.Text>
  /**
   * @optional
   */
}

const Badge = ({
  children,
  size = defaultSize,
  visible = true,
  textColor,
  backgroundColor,
  borderRadius,
  ...rest
}: Props) => {
  const { current: opacity } = React.useRef<Animated.Value>(new Animated.Value(visible ? 1 : 0))
  const { fontScale } = useWindowDimensions()

  const isFirstRendering = React.useRef<boolean>(true)

  React.useEffect(() => {
    // Do not run animation on very first rendering
    if (isFirstRendering.current) {
      isFirstRendering.current = false
      return
    }

    Animated.timing(opacity, {
      toValue: visible ? 1 : 0,
      duration: 150,
      useNativeDriver: true,
    }).start()
  }, [visible, opacity])

  const paddingHorizontal = 4

  return (
    <Animated.Text
      testID={"test-id-badge"}
      numberOfLines={1}
      style={[
        {
          opacity,
          backgroundColor,
          color: textColor,
          fontSize: size * 0.5,
          // ...(!theme.isV3 && theme.fonts.regular),
          lineHeight: size / fontScale,
          height: size,
          minWidth: size,
          borderRadius: borderRadius || size / 2,
          paddingHorizontal,
        },
        styles.container,
        // restStyle,
      ]}
      {...rest}
    >
      {children}
    </Animated.Text>
  )
}

export default Badge

const styles = StyleSheet.create({
  container: {
    alignSelf: "flex-end",
    overflow: "hidden",
    textAlign: "center",
    textAlignVertical: "center",
  },
})
