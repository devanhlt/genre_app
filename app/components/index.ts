export * from "./Typography"
export * from "./Button"
export * from "./Icon"
export * from "./Screen"
export * from "./SvgIcon"
export * from "./InputField"
export * from "./BottomModal"
