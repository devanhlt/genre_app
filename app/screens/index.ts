export * from "./WelcomeScreen"
export * from "./LoginScreen/LoginScreen"
export * from "./ErrorScreen/ErrorBoundary"
export * from "./SplashScreen"
export * from "./ConfigurationScreen"
export * from "./DashboardScreen"
export * from "./LoggingScreen/LoggingScreen"
export * from "./LoggingScreen/LoggingDetailScreen"
export * from "./AccountsScreen/AccountScreen"
