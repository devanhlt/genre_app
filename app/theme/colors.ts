const palette = {
  // Color Yellow
  yellow100: "#FFFCF3",
  yellow200: "#FFDD77",
  yellow300: "#FFCD39",
  yellow400: "#FFC107",
  yellow500: "#997404",
  yellow600: "#594402",

  // Color Blue
  blue100: "#F2F5FB",
  blue200: "#7396D2",
  blue300: "#3466BE",
  blue400: "#0140AE",
  blue500: "#012668",
  blue600: "#00163D",

  // Green color
  green100: "#F2F8F2",
  green200: "#73AC73",
  green300: "#339633",
  green400: "#007C00",
  green500: "#004A00",
  green600: "#002B00",

  // Orange color
  orange100: "#FFF8F3",
  orange200: "#F9B77D",
  orange300: "#F79642",
  orange400: "#F57C13",
  orange500: "#934A0B",
  orange600: "#562B07",

  // Red color
  red: "#C5281C",
  red100: "#FBF2F2",
  red200: "#D87373",
  red300: "#C63333",
  red400: "#B80000",
  red500: "#6E0000",
  red600: "#400000",

  // Neutral color
  neutral0: "#FFFFFF",
  neutral100: "#FBFBFB",
  neutral200: "#E5E8ED",
  neutral300: "#BBC3CF",
  neutral400: "#A1ABBB",
  neutral500: "#8693A8",
  neutral600: "#364B6E",
  neutral700: "#001B48",
  neutral800: "#000F27",
  neutral900: "#000000",

  // Black color
  black: "#1d1d1d",
  black100: "#EEF0F3",
  black200: "#E2E5EB",
  black300: "#A2A8B4",
  black400: "#5E6671",
  black450: "#585E69",
  black500: "#3D4148",
  black600: "#212121",
  black050: "#E5EAF0",

  // Color Gradient
  gradientYellow1: ["#FFDD00", "#FFD700", "#FFC50B", "#FBA819", "#F58220"] as (string | number)[],
  gradientYellow2: ["#F58220", "#F6881F", "#F8991D", "#FDB614", "#FFDD00"] as (string | number)[],
  gradientBlue1: ["#44C8F5", "#009EDA", "#0084C9", "#0076BF", "#0072BC"] as (string | number)[],
  gradientBlue2: ["#0072BC", "#0091D2", "#1DACE3", "#3DBFEF", "#44C8F5"] as (string | number)[],

  transparent: "rgba(0, 0, 0, 0)",

  overlay20: "rgba(25, 16, 21, 0.2)",
  overlay50: "rgba(25, 16, 21, 0.5)",
} as const

export const appColors = {
  /**
   * The palette is available to use, but prefer using the name.
   * This is only included for rare, one-off cases. Try to use
   * semantic names as much as possible.
   */
  palette,
  common: {
    /**
     * App UI background
     * Usage for Element container
     */
    appBackground: palette.neutral100,
    /**
     * White UI background
     * Usage for Element container
     */
    bgRed: palette.red,
    /**
     * White UI background
     * Usage for Element container
     */
    bgWhite: palette.neutral0,
    /**
     * Grey UI background
     * Usage for Element container
     */
    bgGrey: palette.neutral100,
    /**
     * High contract background
     * Tooltip container
     */
    bgInverse: palette.neutral800,
    /**
     * Usage for Display / Headline / Title
     * Icon
     * Press state for $character-secondary
     */
    characterPrimary: palette.neutral800,

    /**
     * Usage for Label / Body / Icon
     */
    characterSecondary: palette.neutral700,

    /**
     * Usage for Support / Icon
     * Hover state for $character-secondary
     */
    characterTertiary: palette.neutral600,

    /**
     * Usage for Label / Link
     * Icon
     */
    characterRedDefault: palette.red400,

    /**
     * Usage for hover state for $character-blue-default
     */
    characterBlueHover: palette.blue300,

    /**
     * Usage for press state for $character-blue-default
     */
    characterBluePress: palette.blue500,

    /**
     * Usage for placeholder character
     */
    characterPlaceholder: palette.neutral500,

    /**
     * Usage for character on color
     */
    characterWhite: palette.neutral0,

    /**
     * Usage for disabled state for character
     */
    characterDisabled: palette.neutral400,

    /**
     * Usage for interactable element
     */
    interactiveDefault: palette.blue400,

    /**
     * Usage for hover state for $input-default
     */
    interactiveHover: palette.blue300,

    /**
     * Usage for press state for $input-default
     */
    interactivePress: palette.red500,

    /**
     * Usage for interactable element on color;
     * White interactable container
     */
    interactiveWhite: palette.neutral0,

    /**
     * Usage for disabled state for interactable
     */
    interactiveDisabled: palette.neutral200,

    /**
     * Usage for Border
     */
    borderDefault: palette.neutral400,

    /**
     * Usage for hover state for $border-default
     */
    borderHover: palette.red300,

    /**
     * Usage for focus state for $border-default
     */
    borderFocus: palette.red400,

    /**
     * Usage for press state for $border-default
     */
    borderPress: palette.red500,

    /**
     * Usage for Divider
     */
    borderDivider: palette.neutral200,

    /**
     * Usage for disabled stated for $border-default
     */
    borderDisabled: palette.neutral300,

    /**
     * Usage for Successful alert
     * Positive feedback
     */
    successDefault: palette.green400,

    /**
     * Usage for hover state for $support-success
     */
    successHover: palette.green300,

    /**
     * Usage for press state for $support-success
     */
    successPress: palette.green500,

    /**
     * Usage for Warning alert
     * Caution feedback
     */
    warningDefault: palette.orange400,

    /**
     * Usage for hover state for $support-warning
     */
    warningHover: palette.orange300,

    /**
     * Usage for press state for $support-warning
     */
    warningPress: palette.orange500,

    /**
     * Usage for Error alert
     * Critical feedback
     */
    errorDefault: palette.red400,

    /**
     * Usage for hover state for $support-error
     */
    errorHover: palette.red300,

    /**
     * Usage for press state for $support-error
     */
    errorPress: palette.red500,

    /**
     * Usage for Divider
     */
    miscDivider: palette.neutral200,

    /**
     * Usage for black overlay
     */
  },
  components: {
    button: {
      primary: palette.red,
      primaryHover: palette.red300,
      primaryPress: palette.neutral200,
      secondary: palette.neutral0,
      disabled: palette.neutral200,
    },
    tag: {
      containerGrey: palette.neutral100,
      containerBlue: palette.blue100,
      containerGreen: palette.green100,
      containerOrange: palette.orange100,
      containerRed: palette.red100,
    },
  },
}

export type ColorsType = keyof typeof appColors | keyof typeof appColors.palette
