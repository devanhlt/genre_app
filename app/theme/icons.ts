/**
  Use these icons for Icon UI element.
 */
export const iconSizes = {
  xsmall: 16,
  small: 20,
  medium: 24,
  large: 32,
} as const
