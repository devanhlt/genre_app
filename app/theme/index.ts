export * from "./colors"
export * from "./spacing"
export * from "./typography"
export * from "./timing"
export * from "./shadow"
export * from "./shape"
export * from "./icons"
