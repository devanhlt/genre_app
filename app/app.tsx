/* eslint-disable import/first */
/**
 * Welcome to the main entry point of the app. In this file, we'll
 * be kicking off our app.
 *
 * Most of this file is boilerplate and you shouldn't need to modify
 * it very often. But take some time to look through and understand
 * what is going on here.
 *
 * The app navigation resides in ./app/navigators, so head over there
 * if you're interested in adding screens and navigators.
 */
if (__DEV__) {
  // Load Reactotron configuration in development. We don't want to
  // include this in our production bundle, so we are using `if (__DEV__)`
  // to only execute this in development.
  require("./devtools/ReactotronConfig.ts")
}
import { QueryClientProvider } from "@tanstack/react-query"
import { useFonts } from "expo-font"
import * as Linking from "expo-linking"
import React, { useState } from "react"
import { initialWindowMetrics, SafeAreaProvider } from "react-native-safe-area-context"
import "./i18n"
import "./utils/ignoreWarnings"

import { BottomSheetModalProvider } from "@gorhom/bottom-sheet"
import { GlobalLoading } from "./components/Loading"
import { ToastMessage } from "./components/Toast"
import Config from "./config"
import { useOnlineManager } from "./hooks/useOnlineManager"
import { RootStore, RootStoreProvider, useInitialRootStore } from "./models"
import { AppNavigator, useNavigationPersistence } from "./navigators"
import { SplashScreen } from "./screens"
import { ErrorBoundary } from "./screens/ErrorScreen/ErrorBoundary"
import QueryClientSetup from "./services/react-query-client"
import { customFontsToLoad } from "./theme"
import { delay } from "./utils/delay"
import * as storage from "./utils/storage"
import { useAppState } from "./hooks/useAppState"

export const NAVIGATION_PERSISTENCE_KEY = "NAVIGATION_STATE"

// Web linking configuration
const prefix = Linking.createURL("/")
const config = {
  screens: {
    Login: {
      path: "",
    },
    Welcome: "welcome",
    Demo: {
      screens: {
        DemoShowroom: {
          path: "showroom/:queryIndex?/:itemIndex?",
        },
        DemoDebug: "debug",
        DemoPodcastList: "podcast",
        DemoCommunity: "community",
      },
    },
  },
}

interface AppProps {
  hideSplashScreen: () => Promise<void>
}

/**
 * This is the root component of our app.
 */
function App(props: AppProps) {
  const { hideSplashScreen } = props
  const [isReady, setIsReady] = useState<boolean>(false)

  /**
   * The Online Manager manages the online state within TanStack Query.
   * It can be used to change the default event listeners or to manually change the online .
   */
  useOnlineManager()

  /**
   * AppState can tell you if the app is in the foreground or background, and notify you when the state changes.
   */
  useAppState()

  // Create a client
  const queryClient = QueryClientSetup.getInstance()

  const {
    initialNavigationState,
    onNavigationStateChange,
    isRestored: isNavigationStateRestored,
  } = useNavigationPersistence(storage, NAVIGATION_PERSISTENCE_KEY)

  const [areFontsLoaded] = useFonts(customFontsToLoad)

  const { rehydrated, rootStore } = useInitialRootStore(async (rootStore: RootStore) => {
    // This runs after the root store has been initialized and rehydrated.

    // If your initialization scripts run very fast, it's good to show the splash screen for just a bit longer to prevent flicker.
    // Slightly delaying splash screen hiding for better UX; can be customized or removed as needed,
    // Note: (vanilla Android) The splash-screen will not appear if you launch your app via the terminal or Android Studio. Kill the app and launch it normally by tapping on the launcher icon. https://stackoverflow.com/a/69831106
    // Note: (vanilla iOS) You might notice the splash-screen logo change size. This happens in debug/development mode. Try building the app for release.
    await delay(1000)
    // Hide splash screen
    hideSplashScreen()
    // Get setting data before render main UI
    if (rootStore.authStore.isAuthenticated) {
      await rootStore.settingStore.fetchSettings()
    }
    setIsReady(true)
  })

  // Before we show the app, we have to wait for our state to be ready.
  // In the meantime, don't render anything. This will be the background
  // color set in native by rootView's background color.
  // In iOS: application:didFinishLaunchingWithOptions:
  // In Android: https://stackoverflow.com/a/45838109/204044
  // You can replace with your own loading component if you wish.
  if (!rehydrated || !isNavigationStateRestored || !areFontsLoaded || !isReady) {
    return <SplashScreen />
  }

  // Configs linking
  const linking = {
    prefixes: [prefix],
    config,
  }

  // otherwise, we're ready to render the app
  return (
    <RootStoreProvider value={rootStore}>
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider initialMetrics={initialWindowMetrics}>
          <BottomSheetModalProvider>
            <ErrorBoundary catchErrors={Config.catchErrors}>
              <AppNavigator
                linking={linking}
                initialState={initialNavigationState}
                onStateChange={onNavigationStateChange}
              />
            </ErrorBoundary>
            <GlobalLoading />
            <ToastMessage />
          </BottomSheetModalProvider>
        </SafeAreaProvider>
      </QueryClientProvider>
    </RootStoreProvider>
  )
}

export default App
