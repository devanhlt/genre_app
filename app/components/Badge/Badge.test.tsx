import React from "react"
import { render, screen } from "@testing-library/react-native"

import { appColors } from "app/theme"
import Badge from "./Badge"

describe("<Badge />", () => {
  it("renders correctly", async () => {
    render(
      <Badge textColor={"#fff"} backgroundColor={appColors.common.bgRed} borderRadius={4}>
        Test
      </Badge>,
    )

    const usernameOutput = await screen.findByTestId("test-id-badge")
    expect(usernameOutput).toHaveTextContent("Test")
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
