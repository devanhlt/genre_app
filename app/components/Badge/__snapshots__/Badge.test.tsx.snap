// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders correctly 1`] = `
<Text
  collapsable={false}
  numberOfLines={1}
  style={
    {
      "alignSelf": "flex-end",
      "backgroundColor": "#C5281C",
      "borderRadius": 4,
      "color": "#fff",
      "fontSize": 10,
      "height": 20,
      "lineHeight": 10,
      "minWidth": 20,
      "opacity": 1,
      "overflow": "hidden",
      "paddingHorizontal": 4,
      "textAlign": "center",
      "textAlignVertical": "center",
    }
  }
  testID="test-id-badge"
>
  Test
</Text>
`;
